Sources/FlowCommoniOSFiles/Animation.swift

```diff
@@ -77,49 +84,65 @@ open class Animation: NSObject, CAAnimationDelegate {
         layer.timeOffset = 0.0
         layer.beginTime = 0.0
         let timeSincePause = layer.convertTime(CACurrentMediaTime(), from: nil) - pausedTime
+
+        for keyframeAnimation in keyframeAnimations {
+            keyframeAnimation.timeOffset = 0
+        }
         layer.beginTime = timeSincePause
     }
 
     /// Pauses the animation.
     open func pause() {
-        let pausedTime = layer.convertTime(CACurrentMediaTime(), from: nil)
-        offset(to: pausedTime)
+        offset(to: layer.convertTime(CACurrentMediaTime(), from: nil))
+        layer.speed = 0
     }
 
     /// Resets the animation to time 0.
-    open func reset() {
+    open func reset(notify: Bool = true) {
         CATransaction.suppressAnimations {
             layer.removeAllAnimations()
-            layer.beginTime = 0
-            offset(to: 0)
-
+            
             for keyframeAnimation in keyframeAnimations {
                 layer.setValue(keyframeAnimation.values?.first, forKeyPath: keyframeAnimation.keyPath!)
             }
 
-            addAllAnimations()
+            offset(to: 0)
+            self.addAllAnimations(notify: notify)
             layer.speed = 0
         }
     }
 
     /// Adds all the animations to `layer` so they can be played.
-    private func addAllAnimations() {
+    private func addAllAnimations(notify: Bool = true) {
```

Comment:
I wouldn't use a default value for `notify`. I would specify the notify value explicitly.

Reply:
private func addAllAnimations()